### PR TITLE
fix: GasPriceOracle ownerless

### PR DESCRIPTION
**Pull request by tynes**
**Description**

The gas price oracle on L2 no longer needs to be `Ownable` - the values are instead set by the `op-node` and `SystemConfig` contract on L1

Fixes ENG-2981

### CODE REVIEW

**Review by trianglesphere — APPROVED** [review]

File: packages/contracts-bedrock/contracts/L2/GasPriceOracle.sol [hunk 1]
```diff
@@ -16,7 +15,14 @@ import { L1Block } from "../L2/L1Block.sol";
...
  *         transaction fee will be.
  */
-contract GasPriceOracle is Ownable, Semver {
+contract GasPriceOracle is Semver {
+    /**
+     * @custom:legacy
+     * @custom:spacer _owner
+     * @notice Spacer for backwards compatibility.
+     */
+    uint256 private spacer_0_0_32;
```
Comment (tynes) [inline packages/contracts-bedrock/contracts/L2/GasPriceOracle.sol:24]: ah this should be an address

File: packages/contracts-bedrock/contracts/L2/GasPriceOracle.sol [hunk 2]
```diff
@@ -16,7 +15,14 @@ import { L1Block } from "../L2/L1Block.sol";
...
  *         transaction fee will be.
  */
-contract GasPriceOracle is Ownable, Semver {
+contract GasPriceOracle is Semver {
+    /**
+     * @custom:legacy
+     * @custom:spacer _owner
+     * @notice Spacer for backwards compatibility.
+     */
+    address private spacer_0_0_32;
```
Comment (smartcontracts) [inline packages/contracts-bedrock/contracts/L2/GasPriceOracle.sol:24]: Should be 0_0_20 to be exact

**Review by smartcontracts — COMMENTED** [review]
How exactly do these get set now?

File: packages/contracts-bedrock/contracts/L2/GasPriceOracle.sol (hunk 2 above)
Comment (tynes) [reply packages/contracts-bedrock/contracts/L2/GasPriceOracle.sol:24]: ah good catch

Comment (tynes) [reply packages/contracts-bedrock/contracts/L2/GasPriceOracle.sol:24]: forgot to update the name when i updated the type

**Comment by tynes** [comment]
> How exactly do these get set now?

These are set via the `SystemConfig` contract

**Review by maurelian — APPROVED** [review]
This LGTM

**Comment by tynes** [comment]
I believe this should be good to merge now, just looking for an approval from @smartcontracts

**Comment by tynes** [comment]
I fixed the merge conflicts here, this should be good to merge now

**Comment by tynes** [comment]
It looks like there is a consistent diff with the bindings build and the erc20 bytecode that is produced. Going to rebuild and ensure that I am using the exact version of foundry that CI is using

**Comment by tynes** [comment]
Not sure why the devnet would be failing like this:

```
Message status: STATE_ROOT_NOT_PUBLISHED

/home/circleci/project/node_modules/@ethersproject/providers/node_modules/@ethersproject/logger/src.ts/index.ts:269
        const error: any = new Error(message);
                           ^
Error: transaction failed [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (transactionHash="0xf994ccb8d337f887fc7de2e83290322c152016c0d58dff0c8689c984cf022504", transaction={"type":2,"chainId":901,"nonce":1,"maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"maxFeePerGas":{"type":"BigNumber","hex":"0x5968592a"},"gasPrice":null,"gasLimit":{"type":"BigNumber","hex":"0x028c73"},"to":"0x4200000000000000000000000000000000000010","value":{"type":"BigNumber","hex":"0x00"},"data":"0x32b7006d0000000000000000000000007c6b91d9be155a6db01f749217d76ff02a7227f20000000000000000000000000000000000000000000000000de0b6b3a7640000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000","accessList":[],"hash":"0xf994ccb8d337f887fc7de2e83290322c152016c0d58dff0c8689c984cf022504","v":1,"r":"0x23906b5225169633afd3d9a1dee3454c5ad6a54bdf1998c4ad1f57ccebfebe73","s":"0x313cd720c2abc63d7c0da409f8b8b06a31b2eb2814d03c1e53451e25b4968731","from":"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266","confirmations":0}, receipt={"to":"0x4200000000000000000000000000000000000010","from":"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266","contractAddress":null,"transactionIndex":2,"gasUsed":{"type":"BigNumber","hex":"0xa42d"},"logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","blockHash":"0x69f4f1dc184b952e58f8334c08fd36bf3b8ca3706d46c213707f7013b158af34","transactionHash":"0xf994ccb8d337f887fc7de2e83290322c152016c0d58dff0c8689c984cf022504","logs":[],"blockNumber":85,"confirmations":15,"cumulativeGasUsed":{"type":"BigNumber","hex":"0x11c43d"},"effectiveGasPrice":{"type":"BigNumber","hex":"0x5968592a"},"status":0,"type":2,"byzantium":true}, code=CALL_EXCEPTION, version=providers/5.7.1)
    at Logger.makeError (/home/circleci/project/node_modules/@ethersproject/providers/node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
```

**Comment by tynes** [comment]
Calling `withdraw` is failing

```
$ cast 4byte 0x32b7006d
withdraw(address,uint256,uint32,bytes)

```

**Pull request merged**